evm_arithmetization/src/cpu/kernel/constants/mod.rs

```diff
@@ -285,3 +294,28 @@ const CODE_SIZE_LIMIT: [(&str, u64); 3] = [
 
 const MAX_NONCE: (&str, u64) = ("MAX_NONCE", 0xffffffffffffffff);
 const CALL_STACK_LIMIT: (&str, u64) = ("CALL_STACK_LIMIT", 1024);
+
+// Cancun-related constants
+
+pub const BEACON_ROOTS_ADDRESS: (&str, [u8; 20]) = (
+    "BEACON_ROOTS_ADDRESS",
+    hex!("000F3df6D732807Ef1319fB7B8bB8522d0Beac02"),
+);
+
+const HISTORY_BUFFER_LENGTH: (&str, u64) = ("HISTORY_BUFFER_LENGTH", 8191);
+
+pub const BEACON_ROOTS_CONTRACT_CODE: [u8; 106] = hex!("60618060095f395ff33373fffffffffffffffffffffffffffffffffffffffe14604d57602036146024575f5ffd5b5f35801560495762001fff810690815414603c575f5ffd5b62001fff01545f5260205ff35b5f5ffd5b62001fff42064281555f359062001fff015500");
```

Comment:
This is the initcode, not the actual contract bytecode.

Reply:
Ah shoot right, the bytecode is
`0x3373fffffffffffffffffffffffffffffffffffffffe14604d57602036146024575f5ffd5b5f35801560495762001fff810690815414603c575f5ffd5b62001fff01545f5260205ff35b5f5ffd5b62001fff42064281555f359062001fff015500`
Will fix it. Thanks @wborgeaud!